app/routes/surveys/components/SurveyDetail.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import React from 'react';
+import { Link } from 'react-router';
+import Time from 'app/components/Time';
+import { Image } from 'app/components/Image';
+import styles from './surveyDetail.css';
+import type { SurveyEntity } from 'app/reducers/surveys';
+import LoadingIndicator from 'app/components/LoadingIndicator';
+import { DetailNavigation } from '../utils.js';
+import Content from 'app/components/Layout/Content';
+
+type Props = {
+  survey: SurveyEntity,
+  deleteSurvey: number => Promise<*>
+};
+
+const SurveyDetail = (props: Props) => {
+  const { survey, deleteSurvey } = props;
+
+  if (!survey || !survey.event) return <LoadingIndicator />;
```

Comment:
Guess this isn't needed since you use `loadingIndicator` in the route.

Reply:
Perhaps not the `!survey` check, but `survey` is initially passed without an `event` attribute which makes the page crash. Idk if this is the cleanest way to avoid that but it works

Reply:
Yeah. Move all the checks to one place, seems strange to do the "same thing" two places (with different options).